types/pg-query-stream/pg-query-stream-tests.ts

```diff
@@ -3,13 +3,17 @@ import * as pg from 'pg';
 
 const options: QueryStream.Options = {
     highWaterMark: 1000,
-    batchSize: 100
+    batchSize: 100,
 };
 
 const query = new QueryStream('SELECT * FROM generate_series(0, $1) num', [1000000], options);
 
 const pool = new pg.Pool();
 pool.connect((err, client, done) => {
+    if (client == null) {
```

Comment:
IMHO this feels very unnatural, I'd normally use something like `err != null` for checking if there's an error.
I see why it's necessary, but I hope it can be fixed by the overload suggestion.

Reply:
I was simply editing a failing test, which did not already check `err`. I agree that doing that would usually be a more sound way to do this.

Reply:
Oh I definitely understand why it's necessary, currently there's no way for TS to know that `client` is non-null if `error` is nullish (if my previous assumption was correct).

This is just a good example of why I'd prefer overloads, so we can prevent these kind of redundant (and unnatural) tests.